libtransmission/peer-mgr.h

```diff
@@ -252,7 +252,7 @@ class tr_peer_info
         {
             num_consecutive_fruitless_ = {};
         }
-        else
+        else if (is_disconnecting)
         {
             on_fruitless_connection();
         }
```

Comment:
I haven't tested it, but don't we want the opposite condition `if (!is_disconnecting)` here?

Or is "stopping torrent" different from disconnecting?

Reply:
`is_disconnecting` is a different concept from the `torrent_stop` I mentioned in the other thread. It's false by default, set in a variety of different cases by calling `tr_peerMsgs::disconnect_soon()`.

I think it fits nicely with the goal of this PR.